latex/generator-tex.sh

```
@@ -0,0 +1,231 @@
+fi
+
+INPUT="$1"
+[ -f "$INPUT" ] || { echo "Error: File not found: $INPUT"; exit 1; }
+
+command -v mmdc >/dev/null || { echo "mmdc missing (npm i -g @mermaid-js/mermaid-cli)"; exit 1; }
+command -v pandoc >/dev/null || { echo "pandoc missing"; exit 1; }
+# Optional: grayscale conversion (ImageMagick)
+if command -v convert >/dev/null; then
+  GRAYSCALE_OK=1
+else
+  GRAYSCALE_OK=0
```

Shellcheck (0.11.0)

[warning] GRAYSCALE_OK appears unused. Verify use (or export if used externally).

(SC2034)